File: app/routers/amocrm.py
# app/routers/amocrm.py
"""amoCRM integration router — OAuth flow, contacts, leads, pipelines, sync."""

import logging
import os
from datetime import datetime, timedelta
from typing import Optional

from fastapi import APIRouter, Depends, HTTPException, Query, Request
from fastapi.responses import RedirectResponse
from pydantic import BaseModel

from app.services.amocrm_service import (
    AmoCRMAPIError,
    add_note_to_lead,
    build_auth_url,
    create_contact,
    create_lead,
    exchange_code_for_token,
    get_account_info,
    get_contacts,
    get_leads,
    get_pipelines,
    refresh_access_token,
)
from auth_manager import User, get_current_user
from db.integration import async_amocrm_manager, async_audit_logger


logger = logging.getLogger(__name__)

router = APIRouter(prefix="/admin/crm", tags=["crm"])
webhook_router = APIRouter(tags=["amocrm-webhook"])


# ============== Helpers ==============


async def _get_valid_token() -> dict:
    """Get config with a valid access token; auto-refresh if expired.

    Returns full config dict with fresh tokens.
    Raises HTTPException if not connected or refresh fails.
    """
    config = await async_amocrm_manager.get_config_with_secrets()
    if not config or not config.get("access_token"):
        raise HTTPException(status_code=400, detail="amoCRM not connected")

    subdomain = config.get("subdomain", "")

    # Check if token expired
    expires_at = config.get("token_expires_at")
    token_expired = True
    if expires_at:
        if isinstance(expires_at, str):
            try:
                expires_at = datetime.fromisoformat(expires_at)
            except ValueError:
                expires_at = None
        if expires_at:
            # Refresh 5 minutes before expiry
            token_expired = datetime.utcnow() >= expires_at - timedelta(minutes=5)

    if token_expired:
        refresh_token = config.get("refresh_token")
        if not refresh_token:
            raise HTTPException(status_code=400, detail="No refresh token, re-authorize")

        try:
            tokens = await refresh_access_token(
                subdomain=subdomain,
                client_id=config.get("client_id", ""),
                client_secret=config.get("client_secret", ""),
                refresh_token=refresh_token,
                redirect_uri=config.get("redirect_uri", ""),
            )
        except AmoCRMAPIError as e:
            logger.error(f"Token refresh failed: {e}")
            raise HTTPException(status_code=401, detail="Token refresh failed, re-authorize")

        expires_in = tokens.get("expires_in", 86400)
        new_expires = datetime.utcnow() + timedelta(seconds=expires_in)

        await async_amocrm_manager.save_config(
            access_token=tokens["access_token"],
            refresh_token=tokens["refresh_token"],
            token_expires_at=new_expires,
        )

        config["access_token"] = tokens["access_token"]
        config["refresh_token"] = tokens["refresh_token"]
        config["token_expires_at"] = new_expires.isoformat()

    return config


# ============== Pydantic Models ==============


class AmoCRMConfigRequest(BaseModel):
    subdomain: Optional[str] = None
    client_id: Optional[str] = None
    client_secret: Optional[str] = None
    redirect_uri: Optional[str] = None
    sync_contacts: Optional[bool] = None
    sync_leads: Optional[bool] = None
    sync_tasks: Optional[bool] = None
    auto_create_lead: Optional[bool] = None
    lead_pipeline_id: Optional[int] = None
    lead_status_id: Optional[int] = None


class CreateContactRequest(BaseModel):
    name: str
    custom_fields: Optional[list] = None


class CreateLeadRequest(BaseModel):
    name: str
    pipeline_id: Optional[int] = None
    status_id: Optional[int] = None
    contact_id: Optional[int] = None


class AddNoteRequest(BaseModel):
    text: str


# ============== Status & Config ==============


@router.get("/status")
async def crm_status(user: User = Depends(get_current_user)):
    """Quick status: connected? token expired? last sync."""
    config = await async_amocrm_manager.get_config()
    if not config:
        return {
            "connected": False,
            "has_credentials": False,
            "last_sync": None,
        }

    return {
        "connected": config.get("is_connected", False),
        "has_credentials": bool(config.get("client_id")),
        "subdomain": config.get("subdomain"),
        "account_info": config.get("account_info"),
        "contacts_count": config.get("contacts_count", 0),
        "leads_count": config.get("leads_count", 0),
        "last_sync": config.get("last_sync_at"),
    }


@router.get("/config")
async def crm_get_config(user: User = Depends(get_current_user)):
    """Get amoCRM config (secrets masked)."""
    config = await async_amocrm_manager.get_config()
    return {"config": config or {}}


@router.post("/config")
async def crm_save_config(
    request: AmoCRMConfigRequest,
    user: User = Depends(get_current_user),
):
    """Save OAuth credentials and sync settings."""
    kwargs = {k: v for k, v in request.model_dump().items() if v is not None}
    config = await async_amocrm_manager.save_config(**kwargs)

    await async_audit_logger.log(
        action="update",
        resource="amocrm_config",
        user_id=user.username,
        details={"fields": list(kwargs.keys())},
    )
    return {"status": "ok", "config": config}


# ============== OAuth Flow ==============


@router.get("/auth-url")
async def crm_auth_url(request: Request, user: User = Depends(get_current_user)):
    """Build OAuth authorization URL for amoCRM."""
    config = await async_amocrm_manager.get_config_with_secrets()
    if not config or not config.get("client_id"):
        raise HTTPException(status_code=400, detail="Client ID not configured")

    subdomain = config.get("subdomain", "")
    if not subdomain:
        raise HTTPException(status_code=400, detail="Subdomain not configured")

    # Build redirect_uri: prefer saved, otherwise auto-detect
    redirect_uri = config.get("redirect_uri")
    if not redirect_uri:
        port = os.getenv("ORCHESTRATOR_PORT", "8002")
        redirect_uri = f"http://localhost:{port}/admin/crm/oauth-redirect"

    url = build_auth_url(
        subdomain=subdomain,
        client_id=config["client_id"],
        redirect_uri=redirect_uri,
    )
    return {"auth_url": url, "redirect_uri": redirect_uri}


@router.get("/oauth-redirect")
async def crm_oauth_redirect(
    code: Optional[str] = None,
    error: Optional[str] = None,
    referer: Optional[str] = None,
):
    """Server-side OAuth callback. amoCRM redirects here with ?code=...

    Exchanges code for tokens, saves, then redirects browser to admin panel.
    No JWT required — this is called by amoCRM redirect, not by the admin panel.
    """
    if error:
        logger.error(f"amoCRM OAuth error: {error}")
        return RedirectResponse(url="/admin/#/crm?error=oauth_denied")

    if not code:
        return RedirectResponse(url="/admin/#/crm?error=no_code")

    config = await async_amocrm_manager.get_config_with_secrets()
    if not config:
        return RedirectResponse(url="/admin/#/crm?error=no_config")

    subdomain = config.get("subdomain", "")
    client_id = config.get("client_id", "")
    client_secret = config.get("client_secret", "")
    redirect_uri = config.get("redirect_uri", "")
    if not redirect_uri:
        port = os.getenv("ORCHESTRATOR_PORT", "8002")
        redirect_uri = f"http://localhost:{port}/admin/crm/oauth-redirect"

    try:
        tokens = await exchange_code_for_token(
            subdomain=subdomain,
            client_id=client_id,
            client_secret=client_secret,
            code=code,
            redirect_uri=redirect_uri,
        )
    except AmoCRMAPIError as e:
        logger.error(f"amoCRM token exchange failed: {e}")
        return RedirectResponse(url="/admin/#/crm?error=token_exchange_failed")

    # Save tokens
    expires_in = tokens.get("expires_in", 86400)
    expires_at = datetime.utcnow() + timedelta(seconds=expires_in)

    # Try to fetch account info
    account_info = {}
    try:
        account_info = await get_account_info(subdomain, tokens["access_token"])
    except Exception as e:
        logger.warning(f"Could not fetch account info: {e}")

    await async_amocrm_manager.save_config(
        access_token=tokens["access_token"],
        refresh_token=tokens["refresh_token"],
        token_expires_at=expires_at,
        redirect_uri=redirect_uri,
        account_info=account_info,
    )

    await async_amocrm_manager.log_sync(
        direction="incoming",
        entity_type="auth",
        action="connect",
        details={"account": account_info.get("name", "")},
    )

    return RedirectResponse(url="/admin/#/crm?connected=true")


@router.post("/disconnect")
async def crm_disconnect(user: User = Depends(get_current_user)):
    """Clear tokens — disconnect from amoCRM."""
    await async_amocrm_manager.clear_tokens()

    await async_audit_logger.log(
        action="delete",
        resource="amocrm_config",
        resource_id="tokens",
        user_id=user.username,
    )

    await async_amocrm_manager.log_sync(
        direction="outgoing",
        entity_type="auth",
        action="disconnect",
    )

    return {"status": "disconnected"}


@router.post("/test")
async def crm_test_connection(user: User = Depends(get_current_user)):
    """Test connection by fetching account info."""
    config = await _get_valid_token()
    try:
        account = await get_account_info(config["subdomain"], config["access_token"])
        return {"status": "ok", "account": account}
    except AmoCRMAPIError as e:
        raise HTTPException(status_code=e.status_code, detail=e.detail)


@router.post("/refresh-token")
async def crm_force_refresh_token(user: User = Depends(get_current_user)):
    """Force token refresh."""
    config = await async_amocrm_manager.get_config_with_secrets()
    if not config or not config.get("refresh_token"):
        raise HTTPException(status_code=400, detail="No refresh token available")

    try:
        tokens = await refresh_access_token(
            subdomain=config.get("subdomain", ""),
            client_id=config.get("client_id", ""),
            client_secret=config.get("client_secret", ""),
            refresh_token=config["refresh_token"],
            redirect_uri=config.get("redirect_uri", ""),
        )
    except AmoCRMAPIError as e:
        raise HTTPException(status_code=e.status_code, detail=e.detail)

    expires_in = tokens.get("expires_in", 86400)
    expires_at = datetime.utcnow() + timedelta(seconds=expires_in)

    await async_amocrm_manager.save_config(
        access_token=tokens["access_token"],
        refresh_token=tokens["refresh_token"],
        token_expires_at=expires_at,
    )
    return {"status": "ok", "expires_at": expires_at.isoformat()}


# ============== Contacts ==============


@router.get("/contacts")
async def crm_get_contacts(
    page: int = Query(1, ge=1),
    limit: int = Query(50, ge=1, le=250),
    query: Optional[str] = None,
    user: User = Depends(get_current_user),
):
    """List contacts from amoCRM (proxied)."""
    config = await _get_valid_token()
    try:
        data = await get_contacts(config["subdomain"], config["access_token"], page, limit, query)
        return data
    except AmoCRMAPIError as e:
        raise HTTPException(status_code=e.status_code, detail=e.detail)


@router.post("/contacts")
async def crm_create_contact(
    request: CreateContactRequest,
    user: User = Depends(get_current_user),
):
    """Create a contact in amoCRM."""
    config = await _get_valid_token()
    try:
        result = await create_contact(
            config["subdomain"],
            config["access_token"],
            name=request.name,
            custom_fields=request.custom_fields,
        )

        await async_amocrm_manager.log_sync(
            direction="outgoing",
            entity_type="contact",
            action="create",
            details={"name": request.name},
        )

        return result
    except AmoCRMAPIError as e:
        raise HTTPException(status_code=e.status_code, detail=e.detail)


# ============== Leads ==============


@router.get("/leads")
async def crm_get_leads(
    page: int = Query(1, ge=1),
    limit: int = Query(50, ge=1, le=250),
    query: Optional[str] = None,
    user: User = Depends(get_current_user),
):
    """List leads from amoCRM (proxied)."""
    config = await _get_valid_token()
    try:
        data = await get_leads(config["subdomain"], config["access_token"], page, limit, query)
        return data
    except AmoCRMAPIError as e:
        raise HTTPException(status_code=e.status_code, detail=e.detail)


@router.post("/leads")
async def crm_create_lead(
    request: CreateLeadRequest,
    user: User = Depends(get_current_user),
):
    """Create a lead in amoCRM."""
    config = await _get_valid_token()
    try:
        result = await create_lead(
            config["subdomain"],
            config["access_token"],
            name=request.name,
            pipeline_id=request.pipeline_id,
            status_id=request.status_id,
            contact_id=request.contact_id,
        )

        await async_amocrm_manager.log_sync(
            direction="outgoing",
            entity_type="lead",
            action="create",
            details={"name": request.name},
        )

        return result
    except AmoCRMAPIError as e:
        raise HTTPException(status_code=e.status_code, detail=e.detail)


@router.post("/leads/{lead_id}/notes")
async def crm_add_note_to_lead(
    lead_id: int,
    request: AddNoteRequest,
    user: User = Depends(get_current_user),
):
    """Add a note to a lead."""
    config = await _get_valid_token()
    try:
        result = await add_note_to_lead(
            config["subdomain"],
            config["access_token"],
            lead_id=lead_id,
            text=request.text,
        )
        return result
    except AmoCRMAPIError as e:
        raise HTTPException(status_code=e.status_code, detail=e.detail)


# ============== Pipelines ==============


@router.get("/pipelines")
async def crm_get_pipelines(user: User = Depends(get_current_user)):
    """List pipelines with their statuses."""
    config = await _get_valid_token()
    try:
        data = await get_pipelines(config["subdomain"], config["access_token"])
        return data
    except AmoCRMAPIError as e:
        raise HTTPException(status_code=e.status_code, detail=e.detail)


# ============== Sync ==============


@router.post("/sync")
async def crm_sync(user: User = Depends(get_current_user)):
    """Manual sync — fetch counts from amoCRM and update local stats."""
    config = await _get_valid_token()
    subdomain = config["subdomain"]
    access_token = config["access_token"]

    contacts_count = 0
    leads_count = 0

    try:
        contacts_full = await get_contacts(subdomain, access_token, page=1, limit=250)
        contacts_list = (contacts_full or {}).get("_embedded", {}).get("contacts", [])
        contacts_count = len(contacts_list)
    except AmoCRMAPIError:
        pass

    try:
        leads_full = await get_leads(subdomain, access_token, page=1, limit=250)
        leads_list = (leads_full or {}).get("_embedded", {}).get("leads", [])
        leads_count = len(leads_list)
    except AmoCRMAPIError:
        pass

    await async_amocrm_manager.save_config(
        contacts_count=contacts_count,
        leads_count=leads_count,
        last_sync_at=datetime.utcnow(),
    )

    await async_amocrm_manager.log_sync(
        direction="incoming",
        entity_type="sync",
        action="sync",
        details={"contacts": contacts_count, "leads": leads_count},
    )

    return {
        "status": "ok",
        "contacts_count": contacts_count,
        "leads_count": leads_count,
        "synced_at": datetime.utcnow().isoformat(),
    }


@router.get("/sync-log")
async def crm_sync_log(
    limit: int = Query(50, ge=1, le=200),
    user: User = Depends(get_current_user),
):
    """Get sync event log."""
    logs = await async_amocrm_manager.get_sync_logs(limit)
    return {"logs": logs}


# ============== Webhook (public) ==============


@webhook_router.post("/webhooks/amocrm")
async def amocrm_webhook(request: Request):
    """Handle incoming amoCRM webhook.

    amoCRM sends POST with form-encoded data for events like:
    contacts[add], contacts[update], leads[add], leads[status], etc.
    """
    try:
        content_type = request.headers.get("content-type", "")

        if "application/json" in content_type:
            data = await request.json()
        else:
            # Form-encoded
            form = await request.form()
            data = dict(form)

        logger.info(
            f"amoCRM webhook received: {list(data.keys()) if isinstance(data, dict) else 'raw'}"
        )

        await async_amocrm_manager.log_sync(
            direction="incoming",
            entity_type="webhook",
            action="receive",
            details=data if isinstance(data, dict) else {"raw": str(data)[:500]},
        )

        return {"status": "ok"}
    except Exception as e:
        logger.error(f"amoCRM webhook error: {e}")
        return {"status": "error", "detail": str(e)}

File: admin/src/api/demo/amocrm.ts
import type { DemoRoute } from './types'
import { daysAgo, minutesAgo, nowISO } from './store'

let connected = true

const accountInfo = {
  id: 31234567,
  name: 'AI Секретарь Demo',
  subdomain: 'ai-sekretar-demo',
  currency: 'RUB',
  created_at: 1690000000,
}

const mockConfig = () => ({
  id: 1,
  subdomain: 'ai-sekretar-demo',
  client_id: 'a1b2c3d4-e5f6-7890-abcd-ef1234567890',
  client_secret_masked: '••••••••••••',
  redirect_uri: window.location.origin + '/admin/crm/oauth-redirect',
  is_connected: connected,
  sync_contacts: true,
  sync_leads: true,
  sync_tasks: false,
  auto_create_lead: true,
  lead_pipeline_id: 7001,
  lead_status_id: 42000001,
  webhook_url: window.location.origin + '/webhooks/amocrm',
  last_sync_at: connected ? minutesAgo(35) : null,
  contacts_count: connected ? 5 : 0,
  leads_count: connected ? 5 : 0,
  account_info: connected ? accountInfo : {},
  created: daysAgo(14),
  updated: minutesAgo(35),
})

const contacts = [
  { id: 50001, name: 'Иванов Алексей', custom_fields_values: [{ field_id: 1, values: [{ value: '+79991234501' }] }] },
  { id: 50002, name: 'Петрова Мария', custom_fields_values: [{ field_id: 1, values: [{ value: '+79991234502' }] }] },
  { id: 50003, name: 'Сидоров Дмитрий', custom_fields_values: [{ field_id: 1, values: [{ value: '+79991234503' }] }] },
  { id: 50004, name: 'Козлова Анна', custom_fields_values: [{ field_id: 1, values: [{ value: '+79991234504' }] }] },
  { id: 50005, name: 'Новиков Сергей', custom_fields_values: [{ field_id: 1, values: [{ value: '+79991234505' }] }] },
]

const leads = [
  { id: 60001, name: 'Подключение VoIP', price: 15000, pipeline_id: 7001, status_id: 42000001 },
  { id: 60002, name: 'Настройка секретаря', price: 25000, pipeline_id: 7001, status_id: 42000002 },
  { id: 60003, name: 'Интеграция CRM', price: 35000, pipeline_id: 7001, status_id: 42000003 },
  { id: 60004, name: 'Обучение персонала', price: 10000, pipeline_id: 7002, status_id: 42000011 },
  { id: 60005, name: 'Поддержка 6 мес', price: 60000, pipeline_id: 7002, status_id: 42000012 },
]

const pipelines = [
  {
    id: 7001,
    name: 'Продажи',
    sort: 1,
    is_main: true,
    _embedded: {
      statuses: [
        { id: 42000001, name: 'Новая заявка', sort: 1, color: '#99ccff' },
        { id: 42000002, name: 'В работе', sort: 2, color: '#ffff99' },
        { id: 42000003, name: 'Согласование', sort: 3, color: '#ffcc66' },
        { id: 42000004, name: 'Успешно', sort: 4, color: '#ccff66' },
      ],
    },
  },
  {
    id: 7002,
    name: 'Сервис',
    sort: 2,
    is_main: false,
    _embedded: {
      statuses: [
        { id: 42000011, name: 'Обращение', sort: 1, color: '#99ccff' },
        { id: 42000012, name: 'Выполняется', sort: 2, color: '#ffff99' },
        { id: 42000013, name: 'Закрыто', sort: 3, color: '#ccff66' },
      ],
    },
  },
]

const syncLogEntries = [
  { id: 1, direction: 'incoming', entity_type: 'contact', entity_id: 50001, action: 'created', details: null, status: 'success', error_message: null, created: minutesAgo(35) },
  { id: 2, direction: 'incoming', entity_type: 'contact', entity_id: 50002, action: 'created', details: null, status: 'success', error_message: null, created: minutesAgo(34) },
  { id: 3, direction: 'incoming', entity_type: 'lead', entity_id: 60001, action: 'created', details: null, status: 'success', error_message: null, created: minutesAgo(33) },
  { id: 4, direction: 'incoming', entity_type: 'lead', entity_id: 60002, action: 'updated', details: { old_status: 42000001, new_status: 42000002 }, status: 'success', error_message: null, created: minutesAgo(32) },
  { id: 5, direction: 'outgoing', entity_type: 'contact', entity_id: 50003, action: 'created', details: null, status: 'success', error_message: null, created: minutesAgo(31) },
  { id: 6, direction: 'incoming', entity_type: 'lead', entity_id: 60003, action: 'created', details: null, status: 'success', error_message: null, created: minutesAgo(30) },
  { id: 7, direction: 'outgoing', entity_type: 'lead', entity_id: 60004, action: 'updated', details: null, status: 'error', error_message: 'Timeout connecting to amoCRM API', created: minutesAgo(29) },
  { id: 8, direction: 'incoming', entity_type: 'contact', entity_id: 50004, action: 'updated', details: null, status: 'success', error_message: null, created: minutesAgo(28) },
  { id: 9, direction: 'incoming', entity_type: 'lead', entity_id: 60005, action: 'created', details: null, status: 'success', error_message: null, created: minutesAgo(27) },
  { id: 10, direction: 'outgoing', entity_type: 'contact', entity_id: 50005, action: 'created', details: null, status: 'success', error_message: null, created: minutesAgo(26) },
]

export const amocrmRoutes: DemoRoute[] = [
  // GET /admin/crm/config
  {
    method: 'GET',
    pattern: /^\/admin\/crm\/config$/,
    handler: () => ({ config: mockConfig() }),
  },
  // POST /admin/crm/config
  {
    method: 'POST',
    pattern: /^\/admin\/crm\/config$/,
    handler: ({ body }) => {
      const b = body as Record<string, unknown>
      if (b.subdomain !== undefined) Object.assign(accountInfo, { subdomain: b.subdomain })
      return { status: 'ok', config: mockConfig() }
    },
  },
  // GET /admin/crm/status
  {
    method: 'GET',
    pattern: /^\/admin\/crm\/status$/,
    handler: () => ({
      connected,
      has_credentials: true,
      subdomain: 'ai-sekretar-demo',
      account_info: connected ? accountInfo : undefined,
      contacts_count: connected ? contacts.length : 0,
      leads_count: connected ? leads.length : 0,
      last_sync: connected ? minutesAgo(35) : null,
    }),
  },
  // GET /admin/crm/auth-url
  {
    method: 'GET',
    pattern: /^\/admin\/crm\/auth-url$/,
    handler: () => ({
      auth_url: '#demo-oauth',
      redirect_uri: window.location.origin + '/admin/crm/oauth-redirect',
    }),
  },
  // POST /admin/crm/disconnect
  {
    method: 'POST',
    pattern: /^\/admin\/crm\/disconnect$/,
    handler: () => {
      connected = false
      return { status: 'ok' }
    },
  },
  // POST /admin/crm/test
  {
    method: 'POST',
    pattern: /^\/admin\/crm\/test$/,
    handler: () => ({
      status: 'ok',
      account: accountInfo,
    }),
  },
  // POST /admin/crm/refresh-token
  {
    method: 'POST',
    pattern: /^\/admin\/crm\/refresh-token$/,
    handler: () => ({
      status: 'ok',
      expires_at: new Date(Date.now() + 86400000).toISOString(),
    }),
  },
  // GET /admin/crm/contacts
  {
    method: 'GET',
    pattern: /^\/admin\/crm\/contacts$/,
    handler: () => ({ _embedded: { contacts } }),
  },
  // POST /admin/crm/contacts
  {
    method: 'POST',
    pattern: /^\/admin\/crm\/contacts$/,
    handler: ({ body }) => {
      const b = body as { name: string; custom_fields?: { field_id: number; values: { value: string }[] }[] }
      const newContact = { id: 50000 + contacts.length + 1, name: b.name, custom_fields_values: b.custom_fields || [] }
      contacts.push(newContact)
      return newContact
    },
  },
  // GET /admin/crm/leads
  {
    method: 'GET',
    pattern: /^\/admin\/crm\/leads$/,
    handler: () => ({ _embedded: { leads } }),
  },
  // POST /admin/crm/leads
  {
    method: 'POST',
    pattern: /^\/admin\/crm\/leads$/,
    handler: ({ body }) => {
      const b = body as { name: string; pipeline_id?: number; status_id?: number }
      const newLead = {
        id: 60000 + leads.length + 1,
        name: b.name,
        price: 0,
        pipeline_id: b.pipeline_id || 7001,
        status_id: b.status_id || 42000001,
      }
      leads.push(newLead)
      return newLead
    },
  },
  // POST /admin/crm/leads/:id/notes
  {
    method: 'POST',
    pattern: /^\/admin\/crm\/leads\/([^/]+)\/notes$/,
    handler: ({ body }) => {
      const b = body as { text: string }
      return { id: Date.now(), text: b.text, created_at: nowISO() }
    },
  },
  // GET /admin/crm/pipelines
  {
    method: 'GET',
    pattern: /^\/admin\/crm\/pipelines$/,
    handler: () => ({ _embedded: { pipelines } }),
  },
  // POST /admin/crm/sync
  {
    method: 'POST',
    pattern: /^\/admin\/crm\/sync$/,
    handler: () => ({
      status: 'ok',
      contacts_count: contacts.length,
      leads_count: leads.length,
      synced_at: nowISO(),
    }),
  },
  // GET /admin/crm/sync-log
  {
    method: 'GET',
    pattern: /^\/admin\/crm\/sync-log$/,
    handler: () => ({ logs: syncLogEntries }),
  },
]
